engine/consensus/compliance/engine_test.go

```diff
@@ -97,62 +112,69 @@ func (cs *ComplianceSuite) TestBroadcastProposalWithDelay() {
 	block.Header.ProposerID = cs.myID
 	cs.payloadDB[block.ID()] = block.Payload
 
-	// keep a duplicate of the correct header to check against leader
-	header := block.Header
+	cs.Run("should fail with wrong proposer", func() {
+		header := *block.Header
+		header.ProposerID = unittest.IdentifierFixture()
+		err := cs.engine.BroadcastProposalWithDelay(&header, 0)
+		require.Error(cs.T(), err, "should fail with wrong proposer")
+		header.ProposerID = cs.myID
+	})
 
-	// unset chain and height to make sure they are correctly reconstructed
-	block.Header.ChainID = ""
-	block.Header.Height = 0
+	// should fail with changed (missing) parent
+	cs.Run("should fail with changed/missing parent", func() {
+		header := *block.Header
+		header.ParentID[0]++
+		err := cs.engine.BroadcastProposalWithDelay(&header, 0)
+		require.Error(cs.T(), err, "should fail with missing parent")
+		header.ParentID[0]--
+	})
 
-	cs.hotstuff.On("SubmitProposal", block.Header, parent.View).Return().Once()
+	// should fail with wrong block ID (payload unavailable)
+	cs.Run("should fail with wrong block ID", func() {
+		header := *block.Header
+		header.View++
+		err := cs.engine.BroadcastProposalWithDelay(&header, 0)
+		require.Error(cs.T(), err, "should fail with missing payload")
+		header.View--
+	})
 
-	// submit to broadcast proposal
-	err := cs.engine.BroadcastProposalWithDelay(block.Header, 0)
-	require.NoError(cs.T(), err, "header broadcast should pass")
+	// unset chain and height to make sure they are correctly reconstructed
```

Comment:
Could we put [lines 141 - 172](https://github.com/onflow/flow-go/blob/f7b85fbb7c67df694a8b92aa4dc7d96860e54702/engine/consensus/compliance/engine_test.go#L141-L172) into their own `cs.Run`  method to increase the structure of the test?